app/build.gradle

```diff
@@ -32,6 +44,27 @@ android {
     aaptOptions {
         cruncherEnabled false
     }
+
+    sourceSets {
+        androidTest {
+            setRoot "src/androidTest"
+            res.srcDir "src/androidTest/res"
+        }
+    }
+}
+
+project.afterEvaluate {
+    def allTasks = getGradle().getStartParameter().getTaskRequests().toString()
+    if (allTasks.contains("AndroidTest")
+            && !allTasks.contains("generate")
+            && project.property("findapet.environment") != "espresso") {
+        throw new InvalidParameterException("You're trying to run espresso tests but haven't set your environment to espresso.")
+    }
+    if (!allTasks.contains("AndroidTest")
+            && allTasks.toString() != "[]"
```

Comment:
This tricky check is to make sure I don't fall into this block during a Gradle sync, who's task list is an empty array, but still returns `false` to `allTasks.isEmpty`. Not sure why...but this works fine.